### ISSUE TITLE
Basic MNIST example with opt_level O1 and O2: loss doesn't decrease

### ISSUE THREAD

**Issue by sorrge** [NVIDIA/apex#747]
I changed pytorch's MNIST example by adding apex.initialize line. When using opt_level O1 or O2, the train loss doesn't decrease. I tried varying loss_scale, including "dynamic", with no effect.
opt_levels O0 and O3 work normally. My GPU has tensor cores. The model has two convolutions and no batchnorm.

I'm using CUDA 10.1, pytorch 1.4.
Attached is the code to reproduce the issue and the output for all four opt_levels. You can see the loss immediately decreasing in O0 and O3, but not in O1 and O2.

[apex_bug.zip](https://github.com/NVIDIA/apex/files/4302874/apex_bug.zip)

**Comment by Lornatang** [comment]
@sorrge 
Because the MNIST data set is relatively simple, the fitting can be achieved after very simple neural network layer training. Try switching to `CIFAR10`?

**Comment by sorrge** [comment]
In this case, the fitting _is not achieved_ at all in O1 and O2 modes. The MNIST performance stays at the same level as a randomly initialized network. The weights are either not updated, or the updates are random.

**Comment by sorrge** [comment]
I found a mistake in my code: I didn't do .backward() properly. I did
`loss.backward()` instead of
```
with amp.scale_loss(loss, optim) as scaled_loss:
    scaled_loss.backward()
```

With the correct code, everything works as expected.
